src/inference/src/os/mac/mac_system_conf.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include <sys/sysctl.h>
+
+#include <memory>
+#include <vector>
+
+#include "dev/threading/parallel_custom_arena.hpp"
+#include "openvino/runtime/system_conf.hpp"
+#include "streams_executor.hpp"
+
+namespace ov {
+
+CPU::CPU() {
+    _num_threads = parallel_get_max_threads();
+    parse_processor_info_macos(
+                             _processors,
+                             _numa_nodes,
+                             _cores,
+                             _proc_type_table);
+}
+
+int parse_processor_info_macos(int& _processors,
+                               int& _numa_nodes,
+                               int& _cores,
+                               std::vector<std::vector<int>>& _proc_type_table) {
+    uint64_t output = 0;
+    size_t size = sizeof(output);
+
+    _processors = 0;
+    _numa_nodes = 0;
+    _cores = 0;
+
+    if (sysctlbyname("hw.ncpu", &output, &size, NULL, 0) < 0) {
```

Comment:
In 32bits MAC, the output also is 64bits?

Reply:
I think "output" is a point to a buffer, "size" specifies the size of the buffer.

Reply:
I mean 8bytes (64bits) is enough for all MAC platform?

Reply:
I think 8bytes is enough